lib/fs.js

```diff
@@ -196,13 +196,76 @@ function Stats(
   this.ino = ino;
   this.size = size;
   this.blocks = blocks;
-  this.atime = new Date(atim_msec);
-  this.mtime = new Date(mtim_msec);
-  this.ctime = new Date(ctim_msec);
-  this.birthtime = new Date(birthtim_msec);
+  this._atim_msec = atim_msec;
+  this._mtim_msec = mtim_msec;
+  this._ctim_msec = ctim_msec;
+  this._birthtim_msec = birthtim_msec;
 }
 fs.Stats = Stats;
 
+Object.defineProperties(Stats.prototype, {
+  atime: {
+    configurable: true,
+    enumerable: true,
+    get() {
+      return this._atime !== undefined ?
+          this._atime :
+          (this._atime = new Date(this._atim_msec + 0.5));
+    },
```

Comment:
Another idea here would be to use another defineProperty within the getter to replace the value once the date is created...

```js
var m = {};
Object.defineProperty(m, 'a', {
  configurable: true,
  enumerable: true,
  get() {
    const val = 1;
    delete this.a;
    Object.defineProperty(this, 'a', { configurable: true, enumerable: true, value: val } );
    return val;
  }
});
```
The getter is called once, during which time it is deleted, and the static value is set... and will be returned every time after. Should have a slightly better performance profile.